### ISSUE TITLE
New SoundFont GUI interface look proposition

### ISSUE THREAD

**Issue by Mark-Agent003** [LMMS/lmms#3517]
I have been looking at a few plugins, and I think soundfont player needs an update. Here's a rough draft:

![image](https://cloud.githubusercontent.com/assets/25230510/25550259/44cfec2a-2c3d-11e7-919d-382f2fc90b35.png)

I went with pink because of the SF2 logo in the corner.

**Comment by RebeccaDeField** [comment]
@Mark-Agent003 Thank you for your work on this!

There are plans to make a coordinated effort to redesign the Instrument Plugins, just as we are doing here #2831 to match the new default theme. Would you be interested in helping us to create a SoundFont design that follows much of the same guidelines as the Effects Plugins? :smile_cat:

**Comment by Mark-Agent003** [comment]
Correct me if I'm wrong. Do I need to make it look more like the current effects plugins in the sense of making things like the knobs green or do you mean more like tweaking the fonts and gradients to the specifications? (Btw, I used the current (orange) triple oscillator plugin as a reference)

**Comment by Mark-Agent003** [comment]
And yes, I would be interested 👍

**Comment by RebeccaDeField** [comment]
@Mark-Agent003 I know that we have agreed that the plugins do need a more standardized redesign, but I'm not sure there has been an official consensus on what, exactly, those standards should be. Looking at past discussions, I can see that people do like to have some extra style and design elements to differentiate these types of plugins.

**I have come up with three options so far:**
- The same look as the effects plugins + a unique logo
- The same look as the effects plugins + a unique logo + make the knobs match the logo color
- Same look as the effects plugins, without any unique design elements

@Umcaruje @BaraMGB @tresf What do you think? Should I open a separate issue on the tracker so that we can nail this down before we start redesigning?

**Comment by Mark-Agent003** [comment]
Ok great! In my opinion I'd like either of the first two options, but preferably the second. It would just give enough uniqueness to the plugins as opposed to the effects.

**Comment by qnebra** [comment]
@RebeccaDeField First option is good for me, but logos without big visual extravaganza.

**Comment by Umcaruje** [comment]
I would disagree with the instruments looking the same as the effect plugins. It would make the GUI look bland imho. 

I always saw them as works of art that should express how the actual plugin sounds. I'd rather we have general guidelines for plugins that would still allow artistic freedom, like not using too many gradients and keeping it simple, than restraining them like we did for the effect plugins.

**Comment by karmux** [comment]
I agree with Umcaruje. The appearance of instruments should give a similar feeling than the sound they are producing.

**Comment by RebeccaDeField** [comment]
I tend to lean more towards what @budislav [has posted in the past](https://github.com/LMMS/lmms/issues/2404#issuecomment-149056388) in terms of what the instruments should look like. :honeybee: 

That being said, I am willing to compromise and open to other ideas.

> I always saw them as works of art that should express how the actual plugin sounds.

What would you think if we allowed for artistic freedom in the layout, logo, fonts and additional design elements, but use the same colors and gradients?


> It would make the GUI look bland imho.

So essentially what we're looking for is a balance between an exciting ui (letting each plugin have its own unique style to represent and visually distinguish its sound) and good ux (maintaining some consistency so that the user does not become [surprised/confused](https://en.wikipedia.org/wiki/Principle_of_least_astonishment) and understands that these plugins have relation to LMMS.

**Comment by Mark-Agent003** [comment]
>I tend to lean more towards what @budislav has posted in the past in terms of what the instruments should look like. 🐝

That's beautiful 😮 
> @Umcaruje What would you think if we allowed for artistic freedom in the layout, logo, fonts and additional design elements, but use the same colors and gradients?

I like this idea, but I think we should use the same _design_ of knobs (not the same color), same font, and limit gradient usage. However, I think colors shouldn't be limited. I think a great example is the new TripleOscillator: 
![image](https://cloud.githubusercontent.com/assets/25230510/25569469/c71ac262-2ddd-11e7-8dba-8a1fe236ce01.png)

**Comment by budislav** [comment]
> I tend to lean more towards what @budislav has posted in the past in terms of what the instruments should look like. 

Appreciate it @RebeccaDeField

This artwork you guys like very much "The appearance of instruments should give a similar feeling than the sound they are producing." is possible now in this present gui system but in my single window concept definitely not.

By default they must have the same look of ui controls like the rest of software! But if you really want they can make this editable so you can make it completely different but I would never say yes to this, that would be wasting a time, effort, talent, nerves and many other things. First of all they need to be all vector based designs so we have no problem on retina screens, that can be really pain for developers and designers. It is much easier for them to develop and for software to run only one gui control set.

I would say yes to make option for editing this theme so you can change main color for all effects for example blue like now, so you can have all effect with green knobs and everything that have blue color in effect to be green, just change that color and same color everywhere, but only those main colors.

Every internal instrument and sound effect belong to LMMS. They are central part of him, so they must look like one big family, all kids from same parents. They are not external virtual instruments like Zynaddsubfx or Sylenth1, or any other. They need to be different. There are many reasons for this and I am sorry I don't have much time to explain this. Just look at ableton live, bitwig, studio one, renoise, sonar and others. They are following the same principle and that is not stupid. There is no place for artwork like that in single window concept I am sorry. The world of interfaces are changing, it's not 90's anymore. Well, on my concept for surely.

**Comment by Mark-Agent003** [comment]
@budislav  You have a very convincing argument. @RebeccaDeField would you like to create a new issue, for redesigning the instruments?

**Comment by Hussam-Eddin-Alhomsi** [comment]
I totally agree with @budislav 

I think that instruments and effects plug-ins should have similar GUIs so that they look more uniformed and belonging to LMMS.
This doesn't mean that they have to look bland. They can have beautiful GUIs while staying around a main theme.

**Comment by tresf** [comment]
Closing as this has been merged into #3534.  Please feel free to continue discussion about SoundFont Player redesign here, however the issue will remain closed in the tracker in favor of the open meta issue.
